src/main/java/org/openrewrite/java/testing/junit5/RemoveTryCatchBlocksFromUnitTests.java

```diff
@@ -68,53 +67,34 @@ public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration md, Execut
             }
             // find try catch block
             List<Statement> statements = md.getBody().getStatements();
-            for (Statement s : statements) {
-                if (s instanceof J.Try) {
-                    List<J.Try.Catch> catch_ = ((J.Try) s).getCatches();
-                    for (J.Try.Catch c : catch_) {
-                        J.Block block = c.getBody();
-                        List<Statement> catchStatements = block.getStatements();
-                        for (Statement st : catchStatements) {
-                            if (st instanceof J.MethodInvocation) {
-                                if (ASSERT_FAIL_MATCHER.matches((J.MethodInvocation) st)) {
-                                    getCursor().putMessage("KEY", md);
-                                }
-                            }
-                        }
-                    }
-                    break;
-                }
-            }
-
-            return super.visitMethodDeclaration(md, ctx);
-        }
-
-        public J.MethodInvocation build(J.MethodInvocation mi) {
-            Expression argument = mi.getArguments().get(0);
-            if (mi.getArguments().get(0) instanceof J.MethodInvocation){
-                argument = ((J.MethodInvocation) mi.getArguments().get(0)).getSelect();
+            List<Statement> tryStatements = statements.stream().filter(s -> s instanceof J.Try).collect(Collectors.toList());
+            if (tryStatements.size() == 0) {
+                return md;
             }
 
-            maybeRemoveImport("org.junit.Assert");
-            maybeAddImport("org.junit.jupiter.api.Assertions");
-            return JavaTemplate.builder("Assertions.assertDoesNotThrow(#{any()})")
-                    .staticImports("org.junit.jupiter.api.Assertions")
-                    .build()
-                    .apply(getCursor(), mi.getCoordinates().replace(), argument);
-        }
-
-        @Override
-        public J.MethodInvocation visitMethodInvocation(J.MethodInvocation mi, ExecutionContext ctx) {
-            J.MethodInvocation m = super.visitMethodInvocation(mi, ctx);
-            Cursor c = getCursor().dropParentWhile(is -> is instanceof J.Block
-                                                         || is instanceof J.Try.Catch
-                                                         || is instanceof J.Try
-                                                         || !(is instanceof Tree));
-            if (c.getMessage("KEY") != null) {
-                return build(m);
+            for (Statement statement : tryStatements) {
```

Comment:
This loop over only the `try` statements of the method's body looks a bit strange to me. Can you add a test with a method containing some more statements before and / or after the `try` statement to make sure that it also then behaves correctly?

Reply:
While adjusting for this I realized, what if there was a case where there were multiple try-catch blocks that needed to be replaced? Is that a case I should be thinking about? If so, can these kinds of changes be made through a `visitTry` visitor? I keep getting errors when I try making changes with this visitor. Seems like these issues could be stemming from the fact that whats being returned is a method and not a `J.Try`.

Reply:
Sounds like you need to switch from an isomorphic visitor (JavaIsoVisitor) to the non isomorphic JavaVisitor , such that you can return a different type of result.

Reply:
Yes, `visitTry()` with a visitor extending `JavaVisitor` rather than `JavaIsoVisitor` is the way forward. That way `visitTry()` can return any other statement (like `J.MethodInvocation` in your case) and it will automatically replace the `J.Try` statement.